client/core/core.go

```diff
@@ -2704,6 +2727,22 @@ func (c *Core) authDEX(dc *dexConnection) error {
 	if len(updatedAssets) > 0 {
 		c.updateBalances(updatedAssets)
 	}
+
+	// Order, match statuses should now be in sync with the DEX.
+	// Refresh the user's orders for each of the DEX's markets and
+	// log updated locked funds info.
+	dc.refreshMarkets()
```

Comment:
I think `updateBalances`, which calls `refreshUser` and accesses the markets map, needs to come after `refreshMarkets`.